frb_codegen/src/generator/dart/ty.rs

```diff
@@ -51,14 +54,54 @@ impl<'a> TypeDartGenerator<'a> {
     pub fn new(ty: IrType, ir_file: &'a IrFile) -> Self {
         let context = TypeGeneratorContext { ir_file };
         match ty {
-            Primitive(ir) => TypePrimitiveGenerator { ir, context }.into(),
-            Delegate(ir) => TypeDelegateGenerator { ir, context }.into(),
-            PrimitiveList(ir) => TypePrimitiveListGenerator { ir, context }.into(),
-            Optional(ir) => TypeOptionalGenerator { ir, context }.into(),
-            GeneralList(ir) => TypeGeneralListGenerator { ir, context }.into(),
-            StructRef(ir) => TypeStructRefGenerator { ir, context }.into(),
-            Boxed(ir) => TypeBoxedGenerator { ir, context }.into(),
-            EnumRef(ir) => TypeEnumRefGenerator { ir, context }.into(),
+            Primitive(ir) => TypePrimitiveGenerator {
```

Comment:
about `dart_api_class_name` - as suggested before, try not to have such mutable fields, but maybe make it immutable

Reply:
I'll have to pass it in the constructor then. is it ok? Cause it will change all the enum variants above

Reply:
Maybe we can do this:

```rust
let context = TypeGeneratorContext { ir_file };
```

becomes 

```rust
let context = TypeGeneratorContext { ir_file, dart_api_class_name };
```

then no need to modify that many things and is easier to read